udf/event_code_points_to_string/udf.sql

```diff
@@ -0,0 +1,54 @@
+CREATE OR REPLACE FUNCTION udf.event_code_points_to_string(code_points ANY TYPE) AS (
+  CODE_POINTS_TO_STRING(
+    IF(
+      code_points IS NULL,
+      NULL,
+      ARRAY(
+        SELECT
+          CASE
+          WHEN
+            i IS NULL
+          THEN
+            NULL
+            -- Skip invalid chars
+            -- https://en.wikibooks.org/wiki/Unicode/Character_reference/D000-DFFF#endnote_SURROGATE
+          WHEN
+            i >= 55294
+          THEN
+            i + 2 + 2048
+            -- Skip commas
+          WHEN
+            i >= 43
+          THEN
+            i + 2
+            -- Skip double quote
+          WHEN
+            i >= 34
+          THEN
+            i + 1
+          ELSE
+            i
+          END
+        FROM
+          UNNEST(code_points) AS i
+      )
+    )
+  )
+);
+
+SELECT
+  assert_equals("!", udf.event_code_points_to_string([33])),
+  assert_equals("#", udf.event_code_points_to_string([34])),
+  assert_equals("-", udf.event_code_points_to_string([43])),
+  assert_equals(CODE_POINTS_TO_STRING([1, 2]), udf.event_code_points_to_string([1, 2])),
+  assert_equals(
+    CODE_POINTS_TO_STRING(NULL),
+    udf.event_code_points_to_string(CAST(NULL AS ARRAY<INT64>))
+  ),
+  assert_equals(CODE_POINTS_TO_STRING([NULL]), udf.event_code_points_to_string([NULL])),
+  assert_equals(CODE_POINTS_TO_STRING([]), udf.event_code_points_to_string([]));
+
```

Comment:
Optionally, a test for the invalid code points >= 55294 could be added

Reply:
I like the current one to guarantee 1M allowable values - if it ends up being slow we can remove it down the line.